### PR TITLE
Implement logic to add or remove 'skip-changelog' label

### PR DESCRIPTION
**Pull request by JohnathonBowers**
## Description

These changes implement the ability to add or remove the "skip-changelog" label from PRs based on the contents of the "Changelog" section in the PR description and the previous state of the PR.

## Issues

Addresses Issue #1

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -22,12 +22,27 @@ async function run() {
...
+      } catch (error) {
+        console.error(`Error updating label "${skipLabel}" for PR #${prNumber}: ${error.message}`);
+      }
+    }
+  } else {
+    try {
+      // Check if the "skip-changelog" label is present on the PR and remove it
+      await updatePRLabel(owner, repo, prNumber, skipLabel, false);
+    } catch (error) {
+      console.error(`Error updating label "${skipLabel}" for PR #${prNumber}: ${error.message}`);
```
Comment (BigSamu) [inline index.js:45]: Why do we need this else statement? We would never have this case. Or is it to avoid the case when someone manually adds that label in the repo? Also, I feel you are repeating yourself. Code in index.js has to be concise. Here a suggestion where we can introduce modularization

```js
const skipLabel = "skip-changelog";

async function handleSkipLabel(entryMap) {
    if (entryMap["skip"]) {
        if (Object.keys(entryMap).length > 1) {
            throw new CategoryWithSkipOptionError();
        }
        await updateLabelAndLog(true);
    } else {
        await updateLabelAndLog(false);
    }
}

async function updateLabelAndLog(addLabel) {
    try {
        await updatePRLabel(owner, repo, prNumber, skipLabel, addLabel);
        if (addLabel) {
            console.log("No changeset file created or updated.");
        }
    } catch (error) {
        console.error(`Error updating label "${skipLabel}" for PR #${prNumber}: ${error.message}`);
    }
}

// Usage
handleSkipLabel(entryMap);
```

File: utils/githubUtils.js [hunk 2]
```diff
@@ -5,16 +5,16 @@ import {
...
 /**
  * Extracts relevant data from a GitHub Pull Request.
  * @returns {Promise<Object>} A promise that resolves to an object containing details of the pull request.
  * @throws {PullRequestDataExtractionError} Throws a custom error if data extraction fails.
  */
 export const extractPullRequestData = async () => {
   try {
-    // Initialize Octokit client with the GitHub token
-    const octokit = github.getOctokit(GITHUB_TOKEN);
-
```
Comment (BigSamu) [inline utils/githubUtils.js:17]: Ok with this! We should do the same for `createOrUpdateFile()`

File: utils/githubUtils.js [hunk 3]
```diff
@@ -5,16 +5,16 @@ import {
 } from "./customErrors.js";
 import { GITHUB_TOKEN } from "../config/constants.js";
 
+// Initialize Octokit client with the GitHub token
+const octokit = github.getOctokit(GITHUB_TOKEN);
+
```
Comment (BigSamu) [inline utils/githubUtils.js:10]: Thinking that we should continue following a functional programming approach. When I created this code I opt an approach more oriented to isolation and self-containarization. But thinking that this client should be one for each user and the state has to be shared along the whole workflow. In that sense I will do the following:

1) Add function `createOctokitClient()` in gitHubUtils

```js
const createOctokitClient = (token) => {
  return github.getOctokit(token);
};
```
2) Modify functions `extractPullRequestData` and `createOrUpdateFile` to accept an oktokit parameter

```js
export const extractPullRequestData = async (octokit) => {
  // Existing logic...
};

export const createOrUpdateFile = async (
  octokit, owner, repo, path, content, message, branchRef
) => {
  // Existing logic...
};
```

3) Create the const variable `ocktokit` in `index.js` (main file that run workflow`

```js
const octokit = createOctokitClient(GITHUB_TOKEN);

// Use the functions with the dynamically created octokit instance
await extractPullRequestData(octokit);
await createOrUpdateFile(octokit, owner, repo, path, content, message, branchRef);
```

File: index.js (hunk 1 above)
Comment (JohnathonBowers) [reply index.js:45]: Admittedly, it would be rare that we would encounter an edge case like this. However, when I was testing the feature, I started by changing our test PR description from a description that had category entries to one that just had a `- skip` entry. I got the following error in the workflow logs:

```
Run BigSamu/OpenSearch_Change_Set_Create_Action@main
Extracting data for PR #2 in BigSamu/OpenSearch-Dashboards
Found 1 changelog entry.
Error updating label "skip-changelog" for PR #2: Label does not exist
Error updating label "skip-changelog" for PR #2: Label does not exist
File: changelogs/fragments/2.yml updated successfully.
```
Additionally, since it's good practice in general to check if something exists before removing it, I figured I would add the `else` statement as a precaution.

Comment (JohnathonBowers) [reply index.js:45]: Also, I like your suggestion for modularizing the code in `index.js`. I can work on implementing that.

File: utils/githubUtils.js (hunk 2 above)
Comment (BigSamu) [reply utils/githubUtils.js:17]: At last I going to keep this inside each function. Later I will include more modularization. That change was introduced I think in PR #16

File: index.js (hunk 1 above)
Comment (BigSamu) [reply index.js:45]: @JohnathonBowers, This suggestion was already implemented or not? Don't see this updated in this PR.

Comment (JohnathonBowers) [reply index.js:45]: > @JohnathonBowers, This suggestion was already implemented or not? Don't see this updated in this PR.

Yes, I implemented this suggestion in PR #11 .

Comment (JohnathonBowers) [reply index.js:45]: The modularization that @BigSamu mentioned has been implemented in PR #11 .

File: utils/githubUtils.js (hunk 2 above)
Comment (JohnathonBowers) [reply utils/githubUtils.js:17]: Sounds good. I'll mark this conversation as resolved.
